src/apps/cbr/CbrReceiver.cc

```diff
@@ -54,7 +55,7 @@ void CbrReceiver::handleMessage(cMessage *msg)
 
     numReceived_++;
     totFrames_ = pPacket->getNframes(); // XXX this value can be written just once
-    int pktSize = (int)pPacket->getByteLength();
+    int pktSize = (int)pPacket->getSize();
```

Comment:
getSize() is not a Packet function, so I guess this is not a Packet instance. It's suspicious to me that this is not correct. check_and_casting a cMessage to CbrPacket may not be what you want to do here.

Reply:
> check_and_casting a cMessage to CbrPacket may not be what you want to do here
Argh, yes you are right, thanks for spotting that one! 

We'll correct this and afterwards getSize() operates on the size field of the CBR header, which is ok.

(We did not find that bug since currently no simulte simulation within the simulation folder there is no simulation using the CbrReceiver... Therefore, this code is currently untested. I guess it would make sense to either remove this app completely or to add a small test scenario.)

Reply:
We fixed that one and also added two new test scenarios (CBR-UL, CBR-DL) within the "Demo" simulation. Fingerprint tests for those two scenarios were also added.